[NEW] User Status Messages

wreiske opened the pull request:
This pull request adds the ability for users to set a custom status message by editing their profile or using a new `/status` slash command.

Status messages show up in direct messages on the header, as well as in the user info sidebar.


### UPDATE
This pull request has changed recently. Please see https://github.com/RocketChat/Rocket.Chat/pull/12710#issuecomment-473618723

rodrigok commented:
@engelgabriel @thiagosanchz what you guys think about this?

geekgonecrazy commented:
Nice!  Would be great to figure out how to surface this where it could be accessed quicker.  But I think this is great way to do it!  Especially with the slash command which will make it instantly usable even on mobile

Hudell commented:
I had been working on #9353 to try and get it merged for a while, maybe both PRs could be merged together as both seem to have features the other lack.

richardt3 commented:
Looks good - it's something my users keep asking for. Obviously would like to see it in the same place as the current status change, but the slash command makes up for it. Can we get this merged to live pls!

geekgonecrazy commented:
@Hudell how hard would this be?  Are both using close enough approches one could be merged and then the other merged to extend the functionality?

wreiske commented:
I'm not sure. I haven't looked too far into the code for the other system. I think for the sake of getting this in, we might as well merge this (since it includes the slash command to allow mobile status changes). My team can commit to working on the extra elements needed to set status from within the UI, as well as working with the IOS and Android apps to get it implemented there.

mickymiek commented:
What's required to get this in? We're very interested in this feature. We could provide help too if needed :)

fifou17 commented:
when will be merged guys ?? any plan ?

wreiske commented:
This will need to be re-worked since a lot of the surrounding code has been moved around. 

I'm willing to put time towards this again, but only with @geekgonecrazy 's or others OK. I don't want to spend more time on a pull-request that is a no-go. Let me know!

geekgonecrazy commented:
I personally still like this and would like to have it in.  Even if the UI changed a bit after implementation.

> @engelgabriel @thiagosanchz what you guys think about this?

@engelgabriel @thiagosanchz Ping! Still would like thoughts :)

sampaiodiego commented:
a lot will need to be changed code wise though as I'm about to move everything away from meteor packages..

wreiske commented:
> a lot will need to be changed code wise though as I'm about to move everything away from meteor packages..

Let me know when it's safe for me to start working on this again. I saw a commit recently that moved a bunch of stuff out of the meteor packages. I'll plan a weekend to hammer out the status messages if you think the current implementation will be okay as a starting point. 

**Essentially:**
- /status slash command to set status (requires no changes to mobile apps to immediately allow changing of statuses while they implement UI for it)
- Change status on the accounts page
- Display status in the user bar 
- Display status in the user info side panel

Hudell commented:
I'll be working on merging this (along with the other status PR too) next week.

@wreiske if you want to solve the conflicts on this one, it would be a huge help!

wreiske commented:
> I'll be working on merging this (along with the other status PR too) next week.
> 
> @wreiske if you want to solve the conflicts on this one, it would be a huge help!

I've updated to the latest build... Also added an Edit Status modal dialog. Trying to figure out how to push/update this pull request. I may need to just start a new pull-request.

![user-status-messages mp4](https://user-images.githubusercontent.com/6295044/54485612-3e39bc00-4852-11e9-8ca0-9ec3ad9a9ef1.gif)

wreiske commented:
Updated to latest greatest!

wreiske commented:
Added rocket.chat output message when using the status slash command. 

![user-status-message-slash mp4](https://user-images.githubusercontent.com/6295044/54495619-9062f780-48bb-11e9-9aa7-726a64f41fb0.gif)

geekgonecrazy commented on app/api/server/v1/users.js:
```diff
@@ -378,6 +380,7 @@ API.v1.addRoute('users.updateOwnBasicInfo', { authRequired: true }, {
 			email: this.bodyParams.data.email,
 			realname: this.bodyParams.data.name,
 			username: this.bodyParams.data.username,
+			statusMessage: this.bodyParams.data.statusMessage,
```
would be really cool if like setAvatar this was a rest call.  Imagining the ability to set status from music players, and many other much more practical integrations.

wreiske replied:
Like my smart toaster! "toast is done".

I'll get this coded up when I wake up unless someone else wants to give it a whack before then.

wreiske replied:
OK, pushed. @geekgonecrazy please take a look at the changes and let me know if I should change it.

wreiske commented:
![set-status-api](https://user-images.githubusercontent.com/6295044/54871736-a9721980-4d8f-11e9-8b18-caf20d599729.gif)

rodrigok commented:
@wreiske the tests are failing, can you check?

wreiske commented:
> @wreiske the tests are failing, can you check?

Can do. Give me a few...

wreiske commented:
> @wreiske the tests are failing, can you check?

@rodrigok fixed with 8ecbe83. There is something else failing for channels, but I don't think it's related to this.... I could be mistaken.

geekgonecrazy commented:
> Edit: Looks like the build is failing because https://github.com/lovell/sharp-libvips/releases/download/v8.7.0/libvips-8.7.0-linux-x64.tar.gz is returning a 502.

looks to be working again.  Must have been having an issue at github

wreiske commented:
> > Edit: Looks like the build is failing because https://github.com/lovell/sharp-libvips/releases/download/v8.7.0/libvips-8.7.0-linux-x64.tar.gz is returning a 502.
> 
> looks to be working again. Must have been having an issue at github

yeah. tests are passing now for mongo 4-0... looks like they are failing at creating a discussion. Unrelated to my changes i think?

wreiske commented:
Closed in favor of #13933

geekgonecrazy commented:
🤗 thanks for helping with this feature!
